src/common/libzmqutil/zap.c

```diff
@@ -133,23 +120,23 @@ static void zap_cb (flux_reactor_t *r,
                 name,
                 status_text);
 
-        if (!(rep = zmsg_new ()))
+        if (!(rep = mpart_create ()))
             goto done;
-        if (!add_zmsg_nth (rep, req, 0)
-                || !add_zmsg_nth (rep, req, 1)
-                || zmsg_addstr (rep, status_code) < 0
-                || zmsg_addstr (rep, status_text) < 0
-                || zmsg_addstr (rep, user_id) < 0
-                || zmsg_addmem (rep, NULL, 0) < 0) {
+        if (!add_mpart_nth (rep, req, 0)
+            || !add_mpart_nth (rep, req, 1)
```

Comment:
No big deal, but looking at this, I wondered why `add_mpart_nth()` wasn't part of the new `mpart` class, and why it returns `bool` instead of `-1` on error like the other `mpart` functions here. Just a thought :thinking:

Reply:
I realize that zap is the only user of mpart, but those helpers (which existed before) seemed kind of less general than the other helpers in mpart?  I agree they should return 0 or -1.  And they could have better names.  I'll fix that at least!

Reply:
> but those helpers (which existed before) seemed kind of less general than the other helpers in mpart? 

I wasn't familiar enough with the code to make that determination, was mainly just curious (and you answered the question :slightly_smiling_face:).